families/sawtooth_validator_registry/validator_registry/src/validator_registry_tp_verifier.rs

```diff
@@ -0,0 +1,212 @@
+        return Err(ApplyError::InternalError(
+            "Verification report does not contain enclave quote body".to_string(),
+        ));
+    }
+    // The ISV enclave quote body is base 64 encoded
+    let _enclave_quote = verification_report_dict
+        .get("isvEnclaveQuoteBody")
+        .expect("Error reading isvEnclaveQuoteBody from verification report");
+    // The report body should be SHA256(SHA256(OPK)|PPK)
+    let hash_input = format!(
+        "{}{}",
+        originator_public_key_hash.to_uppercase(),
+        signup_info.poet_public_key.to_uppercase()
+    );
+    let _hash_value = sha256(hash_input.as_bytes());
+    // TODO: Quote verification
+    // Verify that the nonce in the verification report matches the nonce in the transaction
+    // payload submitted
+    let nonce = match verification_report_dict.get("nonce") {
+        Some(nonce_present) => nonce_present
+            .as_str()
+            .expect("Error reading nonce as string reference"),
+        None => "",
+    };
+    if nonce != signup_info.nonce {
+        error!(
+            "AVR nonce {} does not match signup info nonce {}",
+            nonce, signup_info.nonce
+        );
+        return Err(ApplyError::InternalError(
+            "AVR nonce doesn't match signup info nonce".to_string(),
+        ));
+    }
+    Ok(())
+}
+
+/// Function to verify if message digest (SHA256 of message) is signed using private key
+/// associated with the public key sent as a input parameter. Accepts message, public key and
+/// signature of the message as input parameters.
+///
+/// Note: Digest of message is calculated using SHA256 algorithm in this function.
+fn verify_message_signature(pub_key: &PKey<Public>, message: &[u8], signature: &[u8]) -> bool {
+    let mut verifier = Verifier::new(MessageDigest::sha256(), pub_key)
+        .expect("Error creating verifier object for SHA256 algortihm");
+    verifier
+        .update(message)
+        .expect("Error updating message to verifier");
+    verifier.verify(signature).expect("Error verifying message")
+}
+
+fn get_config_setting(
+    context: &mut TransactionContext,
+    key: &str,
+) -> Result<Option<String>, ApplyError> {
+    let config_key_address = config_key_to_address(&key);
+    let setting_data = get_state(context, &config_key_address);
+
+    match setting_data {
+        Err(err) => Err(err),
+        Ok(entries) => {
+            let setting: Setting = parse_from(&entries.expect("Error reading entries"))?;
+            for entry in setting.get_entries().iter() {
+                if entry.get_key() == key {
+                    return Ok(Some(entry.get_value().to_string()));
+                }
+            }
+            Ok(None)
+        }
+    }
+}
+
+fn config_key_to_address(key: &str) -> String {
+    let _config_address_padding = config_short_hash(String::new());
+
+    let key_parts: Vec<&str> = key.split('.').collect();
+    if key_parts.len() != (_CONFIG_MAX_KEY_PARTS - 1) {
```

Comment:
This will panic!() if config address has 4 or more parts in it or less than 3 parts. However it's generically handled at line 198 for settings string lesser than 4 parts. Can we follow the approach mentioned in sawtooth documentation to handle setting of any length?